### PR TITLE
Create activity filter on app start

### PR DESCRIPTION
**Pull request by stiege**
This avoids needing to create the filter
for every request.

### CODE REVIEW

**Comment by stiege** [comment]
[https://github.com/stiege/grace_bot/pull/1](https://trello.com/c/EvpFP9Wk/17-https-github-com-stiege-grace-bot-pull-1)

File: GraceBot.Tests/ActivityDefinitionTests.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using NUnit.Framework;
+using Moq;
+using System.Collections.Generic;
+
+namespace GraceBot.Tests
```
Comment (NightRare) [inline GraceBot.Tests/ActivityDefinitionTests.cs]: Awesome work @stiege

File: GraceBot.Tests/ActivityDefinitionTests.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    [TestFixture]
+    public class ActivityDefinitionTests
+    {
+        [Test]
+        public void RunWithBadWordTest()
+        {
+            var testLookup = new Dictionary<string, string>()
+            { { "KEY", "value"} };
```
Comment (stiege) [inline GraceBot.Tests/ActivityDefinitionTests.cs]: Sweet.

File: GraceBot.Tests/AppTests.cs [hunk 3]
```diff
@@ -35,6 +35,7 @@ public void CreateFilterOnStartup()
             var mockFactory = new Mock<IFactory>();
             new App(mockFactory.Object);
             mockFactory.Verify(f => f.GetActivityFilter());
+            mockFactory.Verify(f => f.GetActivityDefinition());
```
Comment (stiege) [inline GraceBot.Tests/AppTests.cs]: Good spotting.

File: GraceBot.Tests/packages.config [hunk 4]
```diff
@@ -3,4 +3,5 @@
   <package id="Castle.Core" version="3.3.3" targetFramework="net46" />
   <package id="Moq" version="4.5.28" targetFramework="net46" />
   <package id="NUnit" version="3.5.0" targetFramework="net46" />
+  <package id="NUnit3TestAdapter" version="3.5.1" targetFramework="net46" />
```
Comment (stiege) [inline GraceBot.Tests/packages.config]: Hmm, legit. Not sure if this is actually supposed to be there.

File: GraceBot.Tests/ActivityDefinitionTests.cs (hunk 1 above)
Comment (stiege) [inline GraceBot.Tests/ActivityDefinitionTests.cs]: 👍

**Pull request merged**